src/game/game.service.js

```diff
@@ -1147,6 +1166,20 @@ angular.module('ngGo.Game.Service', [
       //Validate move and get new position
       var newPosition = this.validateMove(x, y, color);
 
+      // Validate the move is not in the existed children node
+      var existedIndex = validateMoveNotInChildrenNode.call(this, x, y, color);
+      if (existedIndex !== null) {
+        // Push new position
+        pushPosition.call(this, newPosition);
+        // Remember the path
+        this.node.rememberedPath = existedIndex;
+        // Change the current node pointer
+        this.node = this.node.children[existedIndex];
+        // Advance the path
+        this.path.advance(existedIndex);
+        return true;
+      }
```

Comment:
What was the reason this check was implemented? I'm trying to figure out the scenario where this would occur and what's happening (sorry been a while since I've worked on ngGo).